libs/services/src/SlippageService/SlippageServiceMain.spec.ts

```diff
@@ -1,47 +1,445 @@
+      // GIVEN: No prices available
+      getUsdPrices.mockResolvedValue([]);
+
+      // WHEN: Get slippage
+      const result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the max slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`one of the tokens is volatile`, async () => {
+      getUsdPrices.mockImplementation(
+        async (chainId, tokenAddress, interval) => {
+          if (tokenAddress === baseTokenAddress) {
+            // GIVEN: One token is volatile
+            return POINTS_VOLATILITY_ZERO;
+          } else {
+            // GIVEN: The other token is very volatile
+            return POINTS_WITH_HIGH_VOLATILITY;
+          }
+        }
+      );
+
+      // WHEN: Get slippage
+      let result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+
+      // WHEN: Get slippage (with the tokens inverted)
+      result = await slippageService.getSlippageBps({
+        chainId,
+        quoteTokenAddress,
+        baseTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage too
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`one of the tokens has no prices available`, async () => {
+      getUsdPrices.mockImplementation(
+        async (chainId, tokenAddress, interval) => {
+          if (tokenAddress === baseTokenAddress) {
+            // GIVEN: One token is volatile
+            return POINTS_VOLATILITY_ZERO;
+          } else {
+            // GIVEN: The other token is not
+            return null;
+          }
+        }
+      );
+
+      // WHEN: Get slippage
+      let result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+
+      // WHEN: Get slippage (with the tokens inverted)
+      result = await slippageService.getSlippageBps({
+        chainId,
+        quoteTokenAddress,
+        baseTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage too
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`if the prices change a lot`, async () => {
+      // GIVEN: The prices don't change at all
+      getUsdPrices.mockResolvedValue(POINTS_WITH_HIGH_VOLATILITY);
+
+      // WHEN: Get slippage
+      const result = await slippageService.getSlippageBps({
+        chainId,
+        baseTokenAddress,
+        quoteTokenAddress,
+      });
+
+      // THEN: We get the maximum slippage
+      expect(result).toBe(MAX_SLIPPAGE);
+    });
+
+    it(`if the asset is volatile`, async () => {
+      // GIVEN: The prices have high volatility
+      getUsdPrices.mockResolvedValue(getPoints([100, 100.02, 100.04, 100.06]));
```

Comment:
0.02% is high?

Reply:
yes, its 0.02% every 5min

According to the calculation, we get `224 BPS`  which I think is volatile